agimus_controller/agimus_controller/trajectories/sine_wave_cartesian_space.py

```diff
@@ -15,48 +18,117 @@ class SinusWaveCartesianSpace(TrajectoryBase):
+        self.amp = np.array(sine_wave_params.amplitude)
+        self.w = np.array(sine_wave_params.pulsation)
         self.w_q = w_q
         self.w_qdot = w_qdot
         self.w_qddot = w_qddot
         self.w_robot_effort = w_robot_effort
         self.w_pose = w_pose
+        self.mask = mask
+        self.ik_q = None
 
-    def get_traj_point_at_t(self, t: np.float64) -> WeightedTrajectoryPoint:
-        quint, _, _ = self.quint_traj.get_value_at_t(t)
-        w = self.w
-        sin_wt = np.sin(w * t)
-        # TODO Implement inv kyn to retrieve correct references
-        self.q = self.q0
-        self.dq = self.dq
-        self.ddq = self.ddq
-        for i in [0, 1]:
-            ee_pose = (
-                self.get_end_effector_pose_from_q(self.q0) + self.amp * quint * sin_wt
+    def initialize(self, pin_model: pin.Model, q0: np.ndarray) -> None:
+        """Initialize the trajectory generator."""
+        super().initialize(pin_model, q0)
+        self.ik_q = self.q0.copy()
+        self.ee_init_pos = self.get_end_effector_pose_from_q_as_se3(self.q0)
+
+    def inverse_kinematics(
+        self,
+        ee_des_pos: pin.SE3,
+        ee_des_vel: np.ndarray,
+        precision=1e-3,
+        it_max=10000,
+    ) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+        """Compute the inverse kinematics of the robot to reach the desired end effector pose."""
+        i = 0
+        success = False
+        while True:
+            self.ik_ee_pose = self.get_end_effector_pose_from_q_as_se3(self.ik_q)
+            dMi = ee_des_pos.actInv(self.ik_ee_pose)
+            error = pin.log(dMi).vector[self.mask]
+            if norm(error) < precision:
+                success = True
+                break
+            if i > it_max:
+                break
+
+            pin.computeJointJacobians(self.pin_model, self.pin_data, self.ik_q)
+            Jee = pin.getFrameJacobian(
+                self.pin_model,
+                self.pin_data,
+                self.ee_frame_id,
+                pin.ReferenceFrame.LOCAL,
+            )[self.mask, :]
+            dq = -Jee.T @ solve(Jee @ Jee.T, error)
+            self.ik_q[:] = pin.integrate(self.pin_model, self.ik_q, dq)
+            i += 1
+
+        if not success:
+            error_msgs = (
+                f"Inverse kinematics 6D failed to converge with error: "
+                f"{self.ik_error}. Number of iteration: {i}"
             )
+            raise RuntimeError(error_msgs)
+
+        pin.forwardKinematics(self.pin_model, self.pin_data, self.ik_q)
+        pin.updateFramePlacement(self.pin_model, self.pin_data, self.ee_frame_id)
+        pin.computeJointJacobians(self.pin_model, self.pin_data, self.ik_q)
+        Jee = pin.getFrameJacobian(
+            self.pin_model,
+            self.pin_data,
+            self.ee_frame_id,
+            pin.ReferenceFrame.LOCAL_WORLD_ALIGNED,
+        )[self.mask, :]
+        dq = -Jee.T @ solve(Jee @ Jee.T, ee_des_vel[self.mask])
```

Comment:
After discussing with @TheoMF, the wrong sign is here.
```suggestion
        dq = Jee.T @ solve(Jee @ Jee.T, ee_des_vel[self.mask])
```